examples/lfn/lfn.py

```diff
@@ -0,0 +1,188 @@
+  """
+  Create a new material and assign it to the active object. "name" should be the
+  name of a material that has been previously loaded using load_materials.
+  """
+  # Figure out how many materials are already in the scene
+  mat_count = len(bpy.data.materials)
+
+  # Create a new material; it is not attached to anything and
+  # it will be called "Material"
+  bpy.ops.material.new()
+
+  # Get a reference to the material we just created and rename it;
+  # then the next time we make a new material it will still be called
+  # "Material" and we will still be able to look it up by name
+  mat = bpy.data.materials['Material']
+  mat.name = 'Material_%d' % mat_count
+  mat.use_nodes = True
+
+  # Attach the new material to the active object
+  # Make sure it doesn't already have materials
+  # import pdb
+  # pdb.set_trace()
+
+  for i in range(len(obj.material_slots)):
+      bpy.ops.object.material_slot_remove({'object': obj})
+
+  # if obj.data.materials:
+  #     obj.data.materials[0] = mat
+  # else:
+  assert len(obj.data.materials) == 0
+  obj.data.materials.append(mat)
+
+  # Find the output node of the new material
+  output_node = None
+  for n in mat.node_tree.nodes:
+    if n.name == 'Material Output':
+      output_node = n
+      break
+
+  # Add a new GroupNode to the node tree of the active material,
+  # and copy the node tree from the preloaded node group to the
+  # new group node. This copying seems to happen by-value, so
+  # we can create multiple materials of the same type without them
+  # clobbering each other
+  group_node = mat.node_tree.nodes.new('ShaderNodeGroup')
+  group_node.node_tree = bpy.data.node_groups[name]
+
+  # Find and set the "Color" input of the new group node
+  for inp in group_node.inputs:
+    if inp.name in properties:
+      inp.default_value = properties[inp.name]
+
+  # Wire the output of the new group node to the input of
+  # the MaterialOutput node
+  mat.node_tree.links.new(
+      group_node.outputs['Shader'],
+      output_node.inputs['Surface'],
+  )
+
+bpy.ops.wm.append(filename=osp.join("./examples/lfn/", "MyMetal.blend", 'NodeTree', "MyMetal"))
+bpy.ops.wm.append(filename=osp.join("./examples/lfn/", "Rubber.blend", 'NodeTree', "Rubber"))
+object = bpy.context.scene.objects[-1]
+
+rand_color = rng.uniform(0, 1, (3,))
+color = (rand_color[0], rand_color[1], rand_color[2], 1)
+add_material('MyMetal', object, Color=color)
+
+# --- Rendering
+logging.info("Rendering the scene ...")
+renderer.save_state(job_dir / "scene.blend")
+data_stack = renderer.render()
+
+# --- Postprocessing
+kb.compute_visibility(data_stack["segmentation"], scene.assets)
+data_stack["segmentation"] = kb.adjust_segmentation_idxs(
+    data_stack["segmentation"],
+    scene.assets,
+    [obj]).astype(np.uint8)
+
+# --- Discard non-used information
+del data_stack["uv"]
+del data_stack["forward_flow"]
+del data_stack["backward_flow"]
+del data_stack["depth"]
+del data_stack["normal"]
```

Comment:
```suggestion
del data_stack["normal"]
del data_stack["object_coordinates"]
```
Added this layer recently.

Reply:
merged